inc/condition.class.php

```diff
@@ -276,153 +286,96 @@ public function showConditionsForItem(PluginFormcreatorConditionnableInterface $
 
       // Get existing conditions for the item
       /** @var CommonDBTM $item */
-      $conditions = $this->getConditionsFromItem($item);
+      $conditions = self::getConditionsFromItem($item);
       foreach ($conditions as $condition) {
          echo $condition->getConditionHtml($item->fields);
       }
    }
 
    /**
-    * return HTML to show a condition line for a question
-    *
-    * @param array $input
+    * get SQL WHERE clause to exclude questions from questions dropdown
     *
-    * @return string HTML to insert in a rendered web page
+    * @param PluginFormcreatorConditionnableInterface $item
+    * @return void
     */
-   public function getConditionHtml(array $input) : string {
-      if ($this->isNewItem()) {
-         $this->getEmpty();
-         $itemtype       = $input['itemtype'];
-         $itemId         = $input['items_id'];
-         $questionId     = '';
-         $show_condition = self::SHOW_CONDITION_EQ;
-         $show_value     = '';
-         $show_logic     = '';
-      } else {
-         $itemtype       = $this->fields['itemtype'];
-         $itemId         = $this->fields['items_id'];
-         $questionId     = $this->fields['plugin_formcreator_questions_id'];
-         $show_condition = $this->fields['show_condition'];
-         $show_value     = $this->fields['show_value'];
-         $show_logic     = $this->fields['show_logic'];
-      }
-      $item              = new $itemtype();
-
-      // Get list of question in the form of the item
-      if (!($item instanceof PluginFormcreatorConditionnableInterface)) {
-         throw new Exception("$itemtype is not a " . PluginFormcreatorConditionnableInterface::class);
-      }
-
-      $rand = mt_rand();
-      $html  = '';
-      $html .= '<tr'
-      . ' data-itemtype="' . self::class . '"'
-      . ' data-id="' . $this->getID() . '"'
-      . '">';
-      $html .= '<td colspan="4">';
-      $html .= '<div class="div_show_condition">';
-
-      // Boolean operator
-      $html.= '<div class="div_show_condition_logic">';
-      $html.= Dropdown::showFromArray('_conditions[show_logic][]',
-            self::getEnumShowLogic(),
-            [
-               'display'               => false,
-               'value'                 => $show_logic,
-               'display_emptychoice'   => false,
-               'rand'                  => $rand,
-            ]);
-      $html.= '</div>';
-
-      // dropdown of questions
-      $form = PluginFormcreatorCommon::getForm();
-      $questionListExclusion = [];
+   public static function getQuestionsExclusion(PluginFormcreatorConditionnableInterface $item) {
```

Comment:
isolate in dedicated method the code to exclude questions from a questions dropdown